File: core/examples/business/refactoring_decision.cpp
/**
 * @file refactoring_decision.cpp
 * @brief ¿Completar refactorización FSK/LoRa Opción B?
 * 
 * Contexto:
 * - Líder técnico con ganas de irse
 * - Cultura "salir del paso"
 * - Refactorización 60% completa (punto sin retorno)
 * - 40% pendiente: actualizar FskScanner + main.cpp
 * 
 * Decisión: ¿Vale la pena terminar?
 */

#include "../../src/core/MonteCarloEngine.h"
#include "../../src/distributions/Distributions.h"
#include "../../src/utils/Analysis.h"
#include <iostream>
#include <iomanip>

using namespace DecisionMaker;
using namespace DecisionMaker::Utils;

/**
 * @brief Escenario de decisión de refactorización
 */
class RefactoringDecisionScenario : public DecisionScenario {
public:
    RefactoringDecisionScenario() : DecisionScenario(
        "Refactorización FSK/LoRa", 
        "¿Completar Strategy Pattern o revertir?"
    ) {}
    
    SimulationResult runSimulation(std::mt19937& rng) const override {
        // Obtener parámetros
        std::string decision = parameters_.getValue<std::string>("decision");
        
        // Distribuciones de incertidumbre
        const auto* compilation_risk_dist = parameters_.getDistribution("compilation_risk");
        const auto* learning_value_dist = parameters_.getDistribution("learning_value");
        const auto* reputation_impact_dist = parameters_.getDistribution("reputation_impact");
        
        double total_value = 0.0;
        double time_invested = 0.0;
        double reputation = 0.0;
        double technical_debt = 0.0;
        bool success = true;
        
        std::uniform_real_distribution<double> uniform(0.0, 1.0);
        
        if (decision == "COMPLETE") {
            // === OPCIÓN A: COMPLETAR REFACTORIZACIÓN ===
            time_invested = 2.5; // 2-4 horas promedio
            
            // Riesgo de compilación (15%)
            double compilation_risk = compilation_risk_dist->sample(rng);
            bool compilation_fails = uniform(rng) < compilation_risk;
            
            if (compilation_fails) {
                // Compilación falla - debugging adicional
                time_invested += 1.0; // +1 hora debugging
                success = uniform(rng) < 0.90; // 90% eventual success
            }
            
            if (success) {
                // Beneficios
                double learning = learning_value_dist->sample(rng);
                reputation = reputation_impact_dist->sample(rng);
                technical_debt = -5.0; // Reduce deuda técnica
                
                total_value = (
                    learning * 7.5 +           // Aprendizaje Strategy Pattern
                    reputation * 8.0 +         // Reputación profesional
                    technical_debt * (-2.0) +  // Reducción deuda técnica
                    10.0                       // Portfolio value
                );
                
                // Penalización por tiempo
                total_value -= time_invested * 0.5;
            } else {
                // Fallo total - peor escenario
                total_value = -10.0;
                reputation = -3.0;
                technical_debt = 0.0; // No cambia
            }
            
        } else if (decision == "REVERT") {
            // === OPCIÓN B: REVERTIR (git reset) ===
            time_invested = 0.08; // 5 minutos
            
            // Siempre exitoso
            success = true;
            
            // Consecuencias
            reputation = -4.0; // Mala reputación por abandonar
            technical_debt = 5.0; // Deuda técnica permanece
            
            total_value = (
                -10.0 +                    // Pérdida trabajo previo
                reputation * 2.0 +         // Impacto reputación
                technical_debt * (-1.5)    // Costo deuda técnica
            );
            
            // Pequeño bonus por tiempo mínimo
            total_value += 2.0;
            
        } else if (decision == "PARTIAL") {
            // === OPCIÓN C: COMMIT PARCIAL + TODO ===
            time_invested = 0.5; // 30 minutos
            
            // Alta probabilidad de éxito
            success = uniform(rng) < 0.95;
            
            if (success) {
                reputation = -2.0; // Trabajo a medias
                technical_debt = 3.0; // Código roto
                
                total_value = (
                    -5.0 +                     // Valor parcial del trabajo
                    reputation * 2.0 +         // Impacto reputación moderado
                    technical_debt * (-2.0) +  // Costo código roto
                    3.0                        // Documentación +valor
                );
            } else {
                total_value = -8.0;
                reputation = -3.0;
            }
        }
        
        // Calcular utilidad final
        double utility = total_value;
        bool good_decision = utility > 0.0;
        
        SimulationResult result(utility, good_decision);
        result.metrics["time_invested"] = time_invested;
        result.metrics["reputation"] = reputation;
        result.metrics["technical_debt"] = technical_debt;
        result.metrics["success"] = success ? 1.0 : 0.0;
        
        return result;
    }
    
    bool validateConfiguration() const override {
        return parameters_.hasParameter("decision") &&
               parameters_.hasParameter("compilation_risk") &&
               parameters_.hasParameter("learning_value") &&
               parameters_.hasParameter("reputation_impact");
    }
    
    std::vector<std::string> getRequiredParameters() const override {
        return {"decision", "compilation_risk", "learning_value", "reputation_impact"};
    }
    
    std::vector<std::string> getProducedMetrics() const override {
        return {"time_invested", "reputation", "technical_debt", "success"};
    }
};

int main() {
    std::cout << "================================================================================\n";
    std::cout << "🎯 ANÁLISIS C++: ¿Completar Refactorización FSK/LoRa?\n";
    std::cout << "================================================================================\n\n";
    
    std::cout << "📋 CONTEXTO:\n";
    std::cout << "   - Líder técnico con ganas de irse\n";
    std::cout << "   - Cultura empresa: 'salir del paso'\n";
    std::cout << "   - Refactorización 60% completa (FskModem creado, Lora limpiado)\n";
    std::cout << "   - Pendiente: FskScanner + main.cpp (40%)\n";
    std::cout << "   - Tiempo estimado: 2-4 horas\n\n";
    
    try {
        std::cout << "🔬 EJECUTANDO SIMULACIONES MONTE CARLO (50,000 iteraciones)...\n\n";
        
        // Distribuciones de incertidumbre
        auto compilation_risk = std::make_unique<UniformDistribution>(0.10, 0.20); // 10-20%
        auto learning_value = std::make_unique<TriangularDistribution>(0.6, 1.0, 0.85);
        auto reputation_impact = std::make_unique<TriangularDistribution>(0.7, 1.0, 0.85);
        
        MonteCarloEngine engine(50000);
        
        // === OPCIÓN A: COMPLETAR ===
        std::cout << "📊 Opción 1: COMPLETAR REFACTORIZACIÓN...\n";
        RefactoringDecisionScenario scenario_complete;
        scenario_complete.getParameters().setValue("decision", std::string("COMPLETE"));
        scenario_complete.getParameters().setDistribution("compilation_risk", compilation_risk->clone());
        scenario_complete.getParameters().setDistribution("learning_value", learning_value->clone());
        scenario_complete.getParameters().setDistribution("reputation_impact", reputation_impact->clone());
        
        auto results_complete = engine.simulate(scenario_complete);
        
        // === OPCIÓN B: REVERTIR ===
        std::cout << "📊 Opción 2: REVERTIR (git reset)...\n";
        RefactoringDecisionScenario scenario_revert;
        scenario_revert.getParameters().setValue("decision", std::string("REVERT"));
        scenario_revert.getParameters().setDistribution("compilation_risk", compilation_risk->clone());
        scenario_revert.getParameters().setDistribution("learning_value", learning_value->clone());
        scenario_revert.getParameters().setDistribution("reputation_impact", reputation_impact->clone());
        
        auto results_revert = engine.simulate(scenario_revert);
        
        // === OPCIÓN C: COMMIT PARCIAL ===
        std::cout << "📊 Opción 3: COMMIT PARCIAL + TODO...\n\n";
        RefactoringDecisionScenario scenario_partial;
        scenario_partial.getParameters().setValue("decision", std::string("PARTIAL"));
        scenario_partial.getParameters().setDistribution("compilation_risk", compilation_risk->clone());
        scenario_partial.getParameters().setDistribution("learning_value", learning_value->clone());
        scenario_partial.getParameters().setDistribution("reputation_impact", reputation_impact->clone());
        
        auto results_partial = engine.simulate(scenario_partial);
        
        // === COMPARACIÓN ===
        std::cout << "================================================================================\n";
        std::cout << "📊 RESULTADOS DEL ANÁLISIS (Monte Carlo C++)\n";
        std::cout << "================================================================================\n\n";
        
        ScenarioComparator comparator;
        comparator.addScenario("Completar Refactorización", results_complete);
        comparator.addScenario("Revertir (git reset)", results_revert);
        comparator.addScenario("Commit Parcial + TODO", results_partial);
        comparator.rankScenarios("mean", false);
        
        std::cout << comparator.generateReport() << "\n\n";
        
        // === ANÁLISIS DETALLADO ===
        std::cout << "🏆 RANKING DE OPCIONES\n";
        std::cout << "--------------------------------------------------------------------------------\n";
        
        auto print_option = [](const std::string& name, const MonteCarloResults& res, int rank) {
            auto time_stats = res.getMetricStats("time_invested");
            auto rep_stats = res.getMetricStats("reputation");
            auto debt_stats = res.getMetricStats("technical_debt");
            auto success_stats = res.getMetricStats("success");
            
            std::cout << "#" << rank << " " << name << "\n";
            std::cout << "    Utilidad promedio: " << std::fixed << std::setprecision(2) 
                      << res.getMean() << "\n";
            std::cout << "    Probabilidad éxito: " << std::setprecision(1) 
                      << success_stats.mean * 100 << "%\n";
            std::cout << "    Tiempo invertido: " << std::setprecision(2) 
                      << time_stats.mean << " horas\n";
            std::cout << "    Impacto reputación: " << std::setprecision(2) 
                      << rep_stats.mean << "\n";
            std::cout << "    Deuda técnica: " << std::setprecision(2) 
                      << debt_stats.mean << "\n";
            
            if (res.getMean() > 10.0) {
                std::cout << "    ⭐⭐⭐ HIGHLY RECOMMENDED\n";
            } else if (res.getMean() > 5.0) {
                std::cout << "    ⭐⭐ RECOMMENDED\n";
            } else if (res.getMean() > 0.0) {
                std::cout << "    ⭐ CONSIDER\n";
            } else {
                std::cout << "    ❌ NOT RECOMMENDED\n";
            }
            std::cout << "\n";
        };
        
        // Ordenar por utilidad
        struct RankedOption {
            std::string name;
            const MonteCarloResults* results;
            double mean;
        };
        
        std::vector<RankedOption> ranked = {
            {"Completar Refactorización", &results_complete, results_complete.getMean()},
            {"Revertir (git reset)", &results_revert, results_revert.getMean()},
            {"Commit Parcial + TODO", &results_partial, results_partial.getMean()}
        };
        
        std::sort(ranked.begin(), ranked.end(), 
            [](const RankedOption& a, const RankedOption& b) { return a.mean > b.mean; });
        
        for (size_t i = 0; i < ranked.size(); ++i) {
            print_option(ranked[i].name, *ranked[i].results, i + 1);
        }
        
        // === RECOMENDACIÓN ===
        std::cout << "================================================================================\n";
        std::cout << "✅ RECOMENDACIÓN C++: " << ranked[0].name << "\n";
        std::cout << "================================================================================\n\n";
        
        if (ranked[0].name.find("Completar") != std::string::npos) {
            std::cout << "🎯 RAZONES PARA COMPLETAR:\n";
            std::cout << "   1. Mayor utilidad esperada: " << std::fixed << std::setprecision(2) 
                      << ranked[0].mean << "\n";
            std::cout << "   2. Código 60% hecho (sunk cost positivo)\n";
            std::cout << "   3. Strategy Pattern → portfolio técnico\n";
            std::cout << "   4. Reduce deuda técnica significativamente\n";
            std::cout << "   5. Mejora reputación profesional\n\n";
            
            std::cout << "⏰ PLAN DE ACCIÓN (2-4 horas):\n";
            std::cout << "   [ ] 1. Actualizar FskScanner.hpp/cpp (15 min)\n";
            std::cout << "   [ ] 2. Actualizar main.cpp FskModem instances (30 min)\n";
            std::cout << "   [ ] 3. Test compilación (15 min)\n";
            std::cout << "   [ ] 4. Fix linker errors (30 min)\n";
            std::cout << "   [ ] 5. Commit 'refactor: Strategy Pattern' (10 min)\n\n";
        } else if (ranked[0].name.find("Revertir") != std::string::npos) {
            std::cout << "⚠️ RAZONES PARA REVERTIR:\n";
            std::cout << "   1. Utilidad esperada: " << std::fixed << std::setprecision(2) 
                      << ranked[0].mean << "\n";
            std::cout << "   2. Mínimo riesgo (5 minutos)\n";
            std::cout << "   3. Código vuelve a estado funcional\n\n";
            
            std::cout << "❌ CONSECUENCIAS:\n";
            std::cout << "   - Pérdida 2-3 horas trabajo\n";
            std::cout << "   - Deuda técnica permanece\n";
            std::cout << "   - Mala reputación (abandonar)\n\n";
        }
        
        std::cout << "================================================================================\n";
        
    } catch (const std::exception& e) {
        std::cerr << "ERROR: " << e.what() << std::endl;
        return 1;
    }
    
    return 0;
}

File: src/decision_maker/analyses/refactoring_decision.py
#!/usr/bin/env python3
"""
Título: Análisis de Decisión - Code Refactoring FSK/LoRa
Propósito: Decidir si completar refactorización de código legacy en proyecto FSK/LoRa
Fecha de Creación: 2025-11-30
Última Actualización: 2025-11-30
Versión: 1.0
Status: Completado

DESCRIPCIÓN:
🎯 Decisión sobre completar refactorización FSK/LoRa Opción B
Contexto del proyecto:
- Líder técnico con intención de irse
- Cultura organizacional "salir del paso"
- Refactorización 60% completa (punto de no retorno)
- 40% pendiente: actualizar FskScanner + main.cpp

PREGUNTA CLAVE:
¿Vale la pena invertir tiempo en terminar la refactorización?

CAMBIOS EN ESTA VERSIÓN (1.0):
- Movido a python/analyses/ directory
- Actualizado import paths
- Análisis completado (decisión tomada)

METODOLOGÍAS USADAS:
- Break-even Analysis (punto de equilibrio esfuerzo/beneficio)
- Decision Trees (secuencia de decisiones)
- Payoff Matrix (comparación de opciones)

RESULTADO FINAL:
Análisis completado - Decisión tomada basada en análisis
Ver results/refactoring/ para detalles

PRÓXIMOS PASOS:
- [ ] N/A - Análisis completado

NOTAS:
- Caso de estudio único para decisión de refactorización
- Factores considerados: deuda técnica, tiempo, cultura organizacional
- Metodología aplicable a futuros casos de refactoring
- Resultados guardados en results/refactoring/
"""

import sys
from pathlib import Path

# Add core to path
sys.path.insert(0, str(Path(__file__).parent.parent / 'core'))

from deep_research_decision_agent import CareerOption, DecisionAnalysisEngine


def create_refactoring_options():
    """Crear las 3 opciones como CareerOptions

    NOTA: salary_expected = valor/beneficio (mayor = mejor)
          NO es tiempo invertido
    """

    # OPCIÓN A: Completar Refactorización (Strategy Pattern)
    option_a = CareerOption(
        name="Completar Refactorización FSK/LoRa (Opción B)",

        # Beneficio: código limpio + aprendizaje + reputación
        salary_expected=7_500_000,  # Alto valor: código limpio, SOLID aplicado, portfolio

        # Probabilidad de éxito
        probability_success=0.85,  # 85% - cambios sencillos, ya 60% hecho

        # Timeline (tiempo invertido en meses)
        timeline_months=0.1,  # <1 día (2-4 horas)

        # Factores (0-10)
        tech_growth=7.5,           # Aprendes Strategy Pattern aplicado
        income_stability=8.0,       # Código más mantenible = menos bugs
        work_life_balance=6.0,      # 2-4h trabajo, debugging posible
        prestige=8.0,               # Dejas código limpio (reputación)
        remote_flexibility=9.0,     # Solo coding, no hardware
        learning_opportunity=7.5,   # SOLID aplicado a firmware real
        career_ceiling=8.0,         # Portafolio con refactorización real

        # Riesgos
        unemployment_risk=0.15,     # 15% - bugs en compilation pueden bloquear
        burnout_risk=0.25,          # 25% - debugging si algo falla
        market_risk=0.10,           # 10% - patrón Strategy siempre útil

        description="""
        Completar refactorización Strategy Pattern:
        1. Actualizar FskScanner.hpp/cpp para usar FskModem* (15 min)
        2. Actualizar main.cpp instancias FskModem (30 min)
        3. Test compilación (15 min)
        4. Fix linker errors si existen (30 min)
        5. Commit refactorización (10 min)

        PROS:
        + Código 60% hecho (sunk cost positivo)
        + FskModem clase limpia creada (600 líneas)
        + Lora clase limpia (FSK removido ~540 líneas)
        + SRP aplicado correctamente
        + Próximo dev hereda código limpio

        CONS:
        - 2-4 horas trabajo (debugging posible)
        - Compilación puede fallar (linker)
        - Testing manual requerido
        """
    )

    # OPCIÓN B: Revertir Todo (git reset)
    option_b = CareerOption(
        name="Revertir Refactorización (git reset --hard)",

        # Beneficio: mínimo (solo recuperas estado anterior)
        salary_expected=2_000_000,  # Bajo valor: código sucio permanece, deuda técnica

        # Probabilidad de éxito
        probability_success=0.99,  # 99% - git reset siempre funciona

        # Timeline
        timeline_months=0.01,  # 5 minutos

        # Factores (0-10)
        tech_growth=2.0,           # No aprendes nada, pérdida
        income_stability=5.0,       # Código vuelve a estado anterior (funcional pero sucio)
        work_life_balance=10.0,     # 5 minutos, sin stress
        prestige=2.0,               # Abandonar trabajo = mala reputación
        remote_flexibility=10.0,    # Un comando git
        learning_opportunity=1.0,   # Cero aprendizaje
        career_ceiling=3.0,         # No mejoras portafolio

        # Riesgos
        unemployment_risk=0.05,     # 5% - prácticamente sin riesgo
        burnout_risk=0.01,          # 1% - mínimo esfuerzo
        market_risk=0.40,           # 40% - deuda técnica permanece

        description="""
        Revertir todos los cambios con git reset:

        PROS:
        + 5 minutos de trabajo
        + Código vuelve a estado funcional conocido
        + Cero riesgo de bugs nuevos

        CONS:
        - Pérdida de 2-3 horas trabajo (FskModem creado, Lora limpiado)
        - Deuda técnica permanece (Lora 1500+ líneas)
        - SRP violación continúa
        - Próximo dev hereda código sucio
        - Mala reputación (abandonar trabajo)
        - NO puedes poner refactorización en CV
        """
    )

    # OPCIÓN C: Commit Parcial + TODO
    option_c = CareerOption(
        name="Commit Trabajo Parcial + TODO.md",

        # Beneficio: trabajo no se pierde, pero código roto
        salary_expected=4_000_000,  # Valor medio: trabajo guardado pero incompleto

        # Probabilidad de éxito
        probability_success=0.95,  # 95% - fácil, solo documentar

        # Timeline
        timeline_months=0.05,  # 30 minutos

        # Factores (0-10)
        tech_growth=5.0,           # Aprendes a documentar refactorización
        income_stability=4.0,       # Código queda ROTO (no compila)
        work_life_balance=8.0,      # 30 min, poco stress
        prestige=4.0,               # Dejas trabajo a medias (mala práctica)
        remote_flexibility=9.0,     # Solo git + markdown
        learning_opportunity=4.0,   # Aprendizaje mínimo
        career_ceiling=4.5,         # Portafolio incompleto

        # Riesgos
        unemployment_risk=0.10,     # 10% - bajo riesgo
        burnout_risk=0.05,          # 5% - poco esfuerzo
        market_risk=0.50,           # 50% - código roto = alto riesgo

        description="""
        Guardar trabajo parcial + documentar:
        1. git add FskModem.hpp/cpp Lora.hpp/cpp (10 min)
        2. Crear TODO.md con pasos pendientes (15 min)
        3. git commit "WIP: FSK refactoring 60% complete" (5 min)

        PROS:
        + Trabajo no se pierde
        + Documentas intención
        + 30 minutos trabajo
        + Próximo dev puede continuar

        CONS:
        - Código NO COMPILA (bloqueante)
        - Próximo dev debe debugear tu trabajo
        - Branch feature no mergeeable
        - Mala práctica (WIP commit)
        - CI/CD roto
        - Equipo no puede usar branch
        """
    )

    return [option_a, option_b, option_c]

def main():
    print("=" * 80)
    print("🎯 ANÁLISIS: ¿Completar Refactorización FSK/LoRa?")
    print("=" * 80)
    print()
    print("📋 CONTEXTO:")
    print("   - Líder técnico con ganas de irse")
    print("   - Cultura empresa: 'salir del paso'")
    print("   - Refactorización 60% completa (FskModem creado, Lora limpiado)")
    print("   - Pendiente: FskScanner + main.cpp (40%)")
    print("   - Tiempo estimado: 2-4 horas")
    print()

    # Crear opciones
    options = create_refactoring_options()

    # Crear engine
    engine = DecisionAnalysisEngine(debug=False)

    # Analizar cada opción con 13 metodologías
    print("🔬 EJECUTANDO 13 METODOLOGÍAS DE ANÁLISIS...")
    print()

    results = []
    for option in options:
        result = engine.analyze_option(option, options)
        results.append({
            'option': option.name,
            'avg_score': result.overall_score,
            'confidence': result.confidence * 100,
            'success_probability': option.probability_success,
            'monte_carlo': result.monte_carlo_score,
            'topsis_rank': result.topsis_rank,
            'pareto': result.pareto_optimal,
            'risk': result.risk_score,
            'recommendation': result.recommendation,
            'description': option.description
        })

    # Ordenar por score
    results.sort(key=lambda x: x['avg_score'], reverse=True)

    # Mostrar resultados
    print()
    print("=" * 80)
    print("📊 RESULTADOS DEL ANÁLISIS")
    print("=" * 80)
    print()

    # Ranking
    print("🏆 RANKING DE OPCIONES")
    print("-" * 80)
    for i, result in enumerate(results, 1):
        stars = "⭐" * min(3, int(result['avg_score'] / 3))
        print(f"#{i} {result['option']}")
        print(f"    Score: {result['avg_score']:.2f}/10 {stars}")
        print(f"    Confianza: {result['confidence']:.1f}%")
        print(f"    Éxito: {result['success_probability']:.0%}")
        print(f"    {result['recommendation']}")
        print()

    # Detalles analíticos
    print("📈 DETALLES ANALÍTICOS")
    print("-" * 80)
    for i, result in enumerate(results, 1):
        print(f"#{i} {result['option']}")
        print(f"    Monte Carlo: {result['monte_carlo']:.2f}")
        print(f"    TOPSIS Rank: #{result['topsis_rank']}")
        print(f"    Pareto Optimal: {result['pareto']}")
        print(f"    Risk Score: {result['risk']:.2f}")
        print()

    # Recomendación
    best = results[0]
    print()
    print("=" * 80)
    print(f"✅ RECOMENDACIÓN: {best['option']}")
    print("=" * 80)
    print()

    if "Completar" in best['option']:
        print("🎯 RAZONES PARA COMPLETAR:")
        print("   1. Ya invertiste 60% del trabajo (sunk cost positivo)")
        print("   2. Código limpio mejora tu reputación profesional")
        print("   3. Strategy Pattern aplicado → portafolio técnico")
        print("   4. Solo 2-4 horas para terminar (manageable)")
        print("   5. Próximo dev hereda código mantenible")
        print()
        print("⏰ PLAN DE ACCIÓN (2-4 horas):")
        print("   [ ] 1. Actualizar FskScanner.hpp/cpp (15 min)")
        print("   [ ] 2. Actualizar main.cpp FskModem instances (30 min)")
        print("   [ ] 3. Test compilación (15 min)")
        print("   [ ] 4. Fix linker errors (30 min)")
        print("   [ ] 5. Commit 'refactor: Strategy Pattern FSK/LoRa' (10 min)")
        print()
    elif "Revertir" in best['option']:
        print("⚠️ RAZONES PARA REVERTIR:")
        print("   1. Cero riesgo (git reset)")
        print("   2. 5 minutos trabajo")
        print("   3. Código vuelve a estado funcional")
        print()
        print("❌ CONSECUENCIAS:")
        print("   - Pérdida 2-3 horas trabajo")
        print("   - Deuda técnica permanece")
        print("   - Mala reputación (abandonar)")
        print()
    else:
        print("⚠️ RAZONES PARA COMMIT PARCIAL:")
        print("   1. Trabajo no se pierde")
        print("   2. 30 minutos trabajo")
        print()
        print("❌ CONSECUENCIAS:")
        print("   - Código NO COMPILA")
        print("   - Branch bloqueado")
        print("   - Próximo dev debe debugear")
        print()

    print("=" * 80)

if __name__ == "__main__":
    main()
